lib/ansible/modules/crypto/luks_device.py

```diff
@@ -251,15 +317,22 @@ def is_luks(self, device):
 
     def run_luks_create(self, device, keyfile):
         # create a new luks container; use batch mode to auto confirm
-        result = self._run_command([self._cryptsetup_bin, 'luksFormat',
-                                    '-q', device, keyfile])
+        label = self._module.params['label']
+        if label is not None:
+            # create luks container v2 with label
+            result = self._run_command([self._cryptsetup_bin, 'luksFormat',
+                                        '--type', 'luks2', '--label', label,
```

Comment:
This will collide with #61522. Also, you should use a similar approach as in there, i.e. don't do an `if` with two copies of `self._run_command()`, but collect the options in a list and then finally call `self._run_command()`.

Reply:
Fixed, thanks! ;-)